src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1797,6 +1815,56 @@ export abstract class AbstractConfigureRecommendedExtensionsAction extends Actio
 			}));
 	}
 
+	protected addRecommendedExtensionToFolder(extensionsFileResource: URI, extensionId: string): TPromise<any> {
+		return this.getOrCreateExtensionsFile(extensionsFileResource)
+			.then(({ content }) => {
+				const jsonContent: IExtensionsConfigContent = json.parse(content) || {};
+				const folderRecommendations = jsonContent.recommendations || [];
+
+				if (folderRecommendations.map(e => e.toLowerCase()).indexOf(extensionId.toLowerCase()) !== -1) {
+					return TPromise.as(null);
+				}
+				folderRecommendations.push(extensionId);
+
+				const folderUnwantedRecommedations = jsonContent.unwantedRecommendations || [];
+				let index = -1;
+				for (let i = 0; i < folderUnwantedRecommedations.length; i++) {
+					if (folderUnwantedRecommedations[i].toLowerCase() === extensionId.toLowerCase()) {
+						index = i;
+						break;
+					}
+				}
```

Comment:
`Array.prototype.findIndex()` is supported in V8, also is `Array.prototype.includes()` but the local TypeScript configuration of the project is erroring out on it, maybe there's a valid reason for it but if there isn't probably something for the team to look into later, will make this kind of things easier to write. (Just a comment, not a requested change)

Reply:
In all the goodness of `map`, `filter`, `some`, I sometimes miss the good old for loop :) 
So I am completely happy with this :) :)